### PR TITLE
Update home page statuses to open, upcoming & closed

### PR DESCRIPTION
**Pull request by djnunez-aot**
-change the status drop down on homepage to handle Upcoming, Open & Closed engagements

-added a time calculation to display upcoming engagements vs open engagements

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by tiagograf-AOT — APPROVED** [review]

**Review by saravanpa-aot — COMMENTED** [review]
is there a ticket number associated with the change?

**Review by saravanpa-aot — CHANGES_REQUESTED** [review]
Please do the requested changes.

**Review by djnunez-aot — COMMENTED** [review]
Changes requested have been made

File: met-web/src/components/landing/LandingComponent.tsx [hunk 1]
```diff
@@ -189,8 +189,9 @@ const LandingComponent = () => {
                                 <MenuItem value={0} sx={{ fontStyle: 'italic', height: '2em' }}>
                                     {''}
                                 </MenuItem>
-                                <MenuItem value={EngagementStatus.Published}>Published</MenuItem>
-                                <MenuItem value={EngagementStatus.Closed}>Closed</MenuItem>
+                                <MenuItem value={EngagementDisplayStatus.Open}>Open</MenuItem>
```
Comment (saravanpa-aot) [inline met-web/src/components/landing/LandingComponent.tsx:192]: can we have constants for Open , Upcoming etc

File: met-web/src/components/landing/LandingComponent.tsx [hunk 2]
```diff
@@ -120,11 +120,11 @@ const LandingComponent = () => {
                         ref={tileBlockRef}
                     >
                         <Grid item xs={12} sm={6} md={4} lg={4}>
-                            <MetLabel sx={{ marginBottom: '2px', display: 'flex' }}>Engagment name</MetLabel>
+                            <MetLabel sx={{ marginBottom: '2px', display: 'flex' }}>Engagement name</MetLabel>
                             <Autocomplete
                                 options={engagementOptions || []}
                                 onInputChange={(_event, newInputValue) => {
-                                    debounceLoadEngagments(newInputValue);
+                                    debounceLoadEngagements(newInputValue);
```
Comment (saravanpa-aot) [inline met-web/src/components/landing/LandingComponent.tsx:127]: good job in renaming... leave the code in a better place than you found it..appreciate it

File: met-web/src/components/engagement/status/index.tsx [hunk 3]
```diff
@@ -42,3 +45,31 @@ export const EngagementStatusChip = ({
...
+            // Engagement is published and open for submission.
+            if (engagementStartDateUTC.isAfter(now)) return <Upcoming preview={preview} />;
+            else {
+                return <Open preview={preview} />;
+            }
+        case SubmissionStatus.Closed:
+            // Engagement is published but it's past the submission date.
+            return <Closed preview={preview} />;
+        default:
+            return null;
```
Comment (saravanpa-aot) [inline met-web/src/components/engagement/status/index.tsx:73]: @tiagograf-AOT any better way to deal this?
or is backend a better place to do this logic?

File: met-web/src/components/landing/LandingComponent.tsx (hunk 1 above)
Comment (djnunez-aot) [reply met-web/src/components/landing/LandingComponent.tsx:192]: I can change the text to constants for you

File: met-web/src/components/engagement/status/index.tsx (hunk 3 above)
Comment (tiagograf-AOT) [reply met-web/src/components/engagement/status/index.tsx:73]: We should be able to use the calculated submission status for this:

                        <EngagementStatusChip submissionStatus={statusName} />

Comment (djnunez-aot) [reply met-web/src/components/engagement/status/index.tsx:73]: I see thank you

File: met-web/src/components/engagement/status/index.tsx [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import { Chip } from '@mui/material';
 import { SubmissionStatus } from 'constants/engagementStatus';
+import { Engagement } from 'models/engagement';
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/status/index.tsx:4]: remove unused imports

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**